packages/arb-token-bridge-ui/src/components/TransferPanel/TransferDisabledDialog.tsx

```diff
@@ -33,12 +39,35 @@ export function TransferDisabledDialog() {
     erc20L1Address: selectedToken?.address,
     chainId: networks.sourceChain.id
   })
+  const [l2ChainIdForTeleport, setL2ChainIdForTeleport] = useState<
+    number | undefined
+  >()
+
+  useEffect(() => {
+    const updateL2ChainIdForTeleport = async () => {
+      if (!isTeleportMode) {
+        return
+      }
+      const { l2ChainId } = await getL2ConfigForTeleport({
+        destinationChainProvider: networks.destinationChainProvider
+      })
```

Comment:
seeing it being called like this i expect `getL2ConfigForTeleport`'s result to be cached but it is not currently

Reply:
Created FS-558